src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -209,4 +224,143 @@ namespace winrt::Microsoft::Terminal::Control::implementation
+    // - currentMatch - the index of the current match (0-based)
+    // Return Value:
+    // - status message
+    winrt::hstring SearchBoxControl::_FormatStatus(int32_t totalMatches, int32_t currentMatch)
+    {
+        if (totalMatches < 0)
+        {
+            return RS_(L"TermControl_Searching");
+        }
+
+        if (totalMatches == 0)
+        {
+            return RS_(L"TermControl_NoMatch");
+        }
+        std::wstringstream ss;
+
+        if (currentMatch < 0 || currentMatch > (MaximumTotalResultsToShowInStatus - 1))
+        {
+            ss << CurrentIndexTooHighStatus;
+        }
+        else
+        {
+            ss << currentMatch + 1;
+        }
+
+        ss << StatusDelimiter;
+
+        if (totalMatches > MaximumTotalResultsToShowInStatus)
+        {
+            ss << TotalResultsTooHighStatus;
+        }
+        else
+        {
+            ss << totalMatches;
+        }
+
+        return ss.str().data();
+    }
+
+    // Method Description:
+    // - Helper method to measure the width of the text block given the text and the font size
+    // Arguments:
+    // - text: the text to measure
+    // - fontSize: the size of the font to measure
+    // Return Value:
+    // - the size in pixels
+    double SearchBoxControl::_TextWidth(winrt::hstring text, double fontSize)
+    {
+        winrt::Windows::UI::Xaml::Controls::TextBlock t;
+        t.FontSize(fontSize);
+        t.Text(text);
+        t.Measure({ FLT_MAX, FLT_MAX });
+        return t.ActualWidth();
+    }
+
+    // Method Description:
+    // - This method tries to predict the maximal size of the status box
+    // by measuring different possible statuses
+    // Return Value:
+    // - the size in pixels
+    double SearchBoxControl::_GetStatusMaxWidth()
+    {
+        const auto fontSize = StatusBox().FontSize();
+        const auto maxLength = std::max({ _TextWidth(_FormatStatus(-1, -1), fontSize),
+                                          _TextWidth(_FormatStatus(0, -1), fontSize),
+                                          _TextWidth(_FormatStatus(MaximumTotalResultsToShowInStatus, MaximumTotalResultsToShowInStatus - 1), fontSize),
+                                          _TextWidth(_FormatStatus(MaximumTotalResultsToShowInStatus + 1, MaximumTotalResultsToShowInStatus - 1), fontSize),
+                                          _TextWidth(_FormatStatus(MaximumTotalResultsToShowInStatus + 1, MaximumTotalResultsToShowInStatus), fontSize) });
+
+        return maxLength;
+    }
+
+    // Method Description:
+    // - Formats and sets the status message in the status box.
+    // Increases the size of the box if required.
+    // Arguments:
+    // - totalMatches - total number of matches (search results)
+    // - currentMatch - the index of the current match (0-based)
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::SetStatus(int32_t totalMatches, int32_t currentMatch)

```

Comment:
Would it be better to not resize the search box? Like, figure out the longest status and just always use that width?

Reply:
Yeah I'm confused about the measurement plus the resizing. I thought we made sure we didn't need to resize, by measuring.

Reply:
That's a good point...